src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeFieldInfo.cs

```diff
@@ -46,6 +47,13 @@ internal RuntimeType GetDeclaringTypeInternal()
 
         public override Module Module => GetRuntimeModule();
         public override bool IsCollectible => m_declaringType.IsCollectible;
+
+        public override bool Equals(object? obj) =>
+            ReferenceEquals(this, obj) ||
+            (MetadataUpdater.IsSupported && CacheEquals(obj));
+
+        public override int GetHashCode() =>
+            HashCode.Combine(MetadataToken.GetHashCode(), m_declaringType.GetUnderlyingNativeHandle().GetHashCode());
```

Comment:
The Metadata token property is expensive virtual call here. I have missed this detail. The GetHashCode and Equals should be moved to RtFieldInfo and MdFieldInfo, and changed to directly use the relevant fields - handle field for RtFieldInfo and token field for MdFieldInfo.